### PR TITLE
tuned: Put it in a dialog on the host page

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by stefwalter** [comment]
Can we call the directory ```tuned```?

In addition:

```
DEBUG: Checking for unpackaged file(s): /usr/lib/rpm/check-files /builddir/build/BUILDROOT/cockpit-wip-1.fc23.x86_64
DEBUG: error: Installed (but unpackaged) file(s) found:
DEBUG:    /usr/share/cockpit/performance/button.html
DEBUG:    /usr/share/cockpit/performance/dialog.html
DEBUG:    /usr/share/cockpit/performance/dialog.js
DEBUG:    /usr/share/cockpit/performance/dialog.min.js.gz
DEBUG:    /usr/share/cockpit/performance/manifest.json
DEBUG: RPM build errors:
DEBUG:     Installed (but unpackaged) file(s) found:
DEBUG:    /usr/share/cockpit/performance/button.html
DEBUG:    /usr/share/cockpit/performance/dialog.html
DEBUG:    /usr/share/cockpit/performance/dialog.js
DEBUG:    /usr/share/cockpit/performance/dialog.min.js.gz
DEBUG:    /usr/share/cockpit/performance/manifest.json
```

File: pkg/systemd/index.html [hunk 1]
```diff
@@ -78,6 +78,11 @@
                     <td>BIOS</td>
                     <td id="system_information_bios_text"></td>
                 </tr>
+                <tr id="system-info-performance">
+                    <td>Performance Profile</td>
```
Comment (stefwalter) [inline pkg/systemd/index.html]: Should be translatable.

File: pkg/tuned/dialog.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                   tuned.call('/Tuned', 'com.redhat.tuned.control', 'recommend_profile', []))
+                .done(function(running, active, recommended) {
+                    is_running = running[0][0];
+                    active_profile = active[0][0];
+                    recommended_profile = recommended[0][0];
+                    update();
+                })
+                .fail(function(ex) {
+                    if (ex.name != "org.freedesktop.DBus.Error.ServiceUnknown")
+                        console.warn("error talking to tuned", ex);
```
Comment (stefwalter) [inline pkg/tuned/dialog.js]: Should the dialog be disabled somehow in this case?

Comment (mvollmer) [inline pkg/tuned/dialog.js]: Yes, I agree.

**Comment by mvollmer** [comment]
> Can we call the directory ```tuned```?

But we would still use `name: "performance"` in the manifest and load it as `performance/dialog`, right?

**Comment by stefwalter** [comment]
> But we would still use name: "performance" in the manifest and load it as performance/dialog, right?

I think so, yes.

**Comment by mvollmer** [comment]
>> But we would still use name: "performance" in the manifest and load it as performance/dialog, right?

> I think so, yes.

Ok, done.

File: pkg/tuned/dialog.js (hunk 2 above)
Comment (mvollmer) [inline pkg/tuned/dialog.js]: Done.

File: pkg/systemd/index.html (hunk 1 above)
Comment (mvollmer) [inline pkg/systemd/index.html]: Done.

**Comment by mvollmer** [comment]
Images have been uploaded.

File: test/guest/fedora-22.setup [hunk 3]
```diff
@@ -86,10 +80,7 @@ dnf -y copr enable phatina/storaged
 # We need avocado for some integration tests
 dnf -y -q copr enable lmr/Autotest
```
Comment (dperpeet) [inline test/guest/fedora-22.setup]: this is only needed for avocado I think

File: test/guest/fedora-testing.setup [hunk 4]
```diff
@@ -81,10 +75,7 @@ dnf -y upgrade
 dnf -y -q copr enable lmr/Autotest
```
Comment (dperpeet) [inline test/guest/fedora-testing.setup]: this is only needed for avocado I think

File: test/guest/rhel-7.setup [hunk 5]
```diff
@@ -105,12 +100,7 @@ rm -rf /tmp/dep
 # get autotest
 wget -T 15 -t 4 -O /etc/yum.repos.d/lmr-Autotest-epel-7.repo https://copr.fedoraproject.org/coprs/lmr/Autotest/repo/epel-7/lmr-Autotest-epel-7.repo
```
Comment (dperpeet) [inline test/guest/rhel-7.setup]: this is only needed for avocado I think

**Comment by dperpeet** [comment]
I think it would be good to include the extra avocado depency when we're creating new images anyway: https://github.com/cockpit-project/cockpit/pull/3546/files#diff-1c3a743211985f0361fe807ff32b55edR39
@jscotka for the other distributions, you can use vm-customize instead

**Comment by mvollmer** [comment]
This needs to be made fully optional for Debian unstable and the Atomics.

**Comment by mvollmer** [comment]
> This needs to be made fully optional for Debian unstable and the Atomics.

Let's just ship the code also on Debian and let it say "Tuned is unavailable".

**Comment by mvollmer** [comment]
> I think it would be good to include the extra avocado depency when we're creating new images anyway

This is done in #3547

File: pkg/systemd/host.js [hunk 6]
```diff
@@ -22,6 +22,7 @@ define([
     "base1/cockpit",
     "base1/mustache",
     "domain/operation",
+    "performance/dialog",
```
Comment (dperpeet) [inline pkg/systemd/host.js]: rename to tuned?

File: pkg/systemd/index.html [hunk 7]
```diff
@@ -13,6 +13,7 @@
     <script src="../shell/shell.js"></script>
     <script src="../system/bundle.js"></script>
     <script src="../domain/bundle.js"></script>
+    <script src="../performance/bundle.js"></script>
```
Comment (dperpeet) [inline pkg/systemd/index.html]: rename to tuned?

File: pkg/systemd/host.js [hunk 8]
```diff
@@ -32,7 +33,7 @@ define([
     "base1/bootstrap-datepicker",
     "base1/bootstrap-combobox",
     "base1/patterns",
-], function($, cockpit, Mustache, domain, controls, shell, server, service) {
+], function($, cockpit, Mustache, domain, performance, controls, shell, server, service) {
```
Comment (dperpeet) [inline pkg/systemd/host.js]: for consistency, should we rename this to tuned as well?

File: pkg/systemd/host.js (hunk 6 above)
Comment (stefwalter) [inline pkg/systemd/host.js]: No, the directory and module have different names. That's the way it is with the realmd stuff too.

Comment (dperpeet) [inline pkg/systemd/host.js]: > No, the directory and module have different names. That's the way it is with the realmd stuff too.

ok, noted!

File: pkg/systemd/index.html (hunk 7 above)
Comment (mvollmer) [inline pkg/systemd/index.html]: I'd say no, just as with domain/realmd.

File: pkg/systemd/host.js (hunk 8 above)
Comment (mvollmer) [inline pkg/systemd/host.js]: I'd say no, just as with domain/realmd.

File: pkg/systemd/index.html (hunk 7 above)
Comment (dperpeet) [inline pkg/systemd/index.html]: > I'd say no, just as with domain/realmd.

ok

**Comment by stefwalter** [comment]
I see a dialog like this when i show the dialog for a second time:

screenshot from 2016-01-25 09-21-52

**Comment by dperpeet** [comment]
> I see a dialog like this when i show the dialog for a second time:

How did you get this to happen? (I guess you mean the empty entries.) I can't make that happen here on fedora-23.

**Comment by stefwalter** [comment]
It happens in Firefox.

**Comment by dperpeet** [comment]
> It happens in Firefox.

works for me with Firefox 38.5.0

**Comment by stefwalter** [comment]
Fix for the rendering issue a8afd99eef65d99a04ce3288d3770d41bf154ea1

**Comment by mvollmer** [comment]
> I see a dialog like this when i show the dialog for a second time:

I can't get this to happen, either.  What is the fix?  Removing `dialog` when it is hidden?

**Comment by stefwalter** [comment]
> I can't get this to happen, either. What is the fix? Removing dialog when it is hidden?

Yup.

**Comment by mvollmer** [comment]
> Fix for the rendering issue a8afd99

Thanks, merged.

**Comment by andreasn** [comment]
The items in the list needs cursor: pointer; when hovered.

**Comment by andreasn** [comment]
and also a entry background of #D5ECF9 when hovered in addition to the pointer change.

**Comment by andreasn** [comment]
This will fix it https://github.com/andreasn/cockpit/tree/tuned-dialog-hover

**Comment by dperpeet** [comment]
https://github.com/andreasn/cockpit/commit/bb1f227fd0ab46b21bf83a9ef417667027891922
is the fixup by @andreasn

**Comment by mvollmer** [comment]
Tests are broken..

File: pkg/tuned/dialog.html [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  <div id="tuned-dialog" class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h4 class="modal-title" translatable="yes">Change Performance Profile</h4>
+      </div>
+      <div class="modal-body">
+        <div class="list-group dialog-list">
+          {{#Profiles}}
+          <div class="list-group-item" data-profile="{{profile}}">
+            {{Title}}{{#recommended}}<em> (recommended)</em>{{/recommended}}</br>
```
Comment (dperpeet) [inline pkg/tuned/dialog.html]: I think this should be `<br>` instead

**Comment by mvollmer** [comment]
> Tests are broken..

Fixed.

**Comment by mvollmer** [comment]
> This will fix it https://github.com/andreasn/cockpit/tree/tuned-dialog-hover

Included, thanks!

File: pkg/tuned/dialog.html (hunk 9 above)
Comment (mvollmer) [inline pkg/tuned/dialog.html]: Yes, ouch.

Comment (mvollmer) [inline pkg/tuned/dialog.html]: Fixed.

**Comment by dperpeet** [comment]
I can remove the avocado dependencies in a follow-up.

**Pull request closed without merging**